### PR TITLE
Check on unconcious dragging out of sleeper

### LINKED ISSUE #7128: Issue Name SLEEPER EXPLOIT SUPER EXPLOITABLE
(WEB REPORT BY: clugehead REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)
> SAME AS ABOVVE
> 
> General description of the issue
> You can click and drag yourself out of a sleeper if sleeping
> 
> What you expected to happen
> To not eject myself
> 
> What actually happened
> It worked, absolute madman
> 
> Steps to reproduce if possible
> Click and drag self out of sleeper when asleep. I was chloralled so test that if it don't work.

### PR DESCRIPTION
**Pull request by Steelcap**
fixes #7128

### CODE REVIEW

File: code/game/machinery/Sleeper.dm [hunk 1]
```diff
@@ -306,6 +306,11 @@
 		return
 	if(!Adjacent(over_location))
 		return
+		
+	var/mob/user = usr
+	if (user.stunned || user.paralysis || user.resting) // If you're too disabled to put someone into it, you're too disabled to pull someone out of it.
```
Comment (9600bauds) [inline code/game/machinery/Sleeper.dm]: Look at mob_helpers.dm, there are procs to check this better. You probably want to use user.isUnconscious(), and check for user.restrained() while you're at it.

**Comment by 9600bauds** [comment]
I kind of meant that isUnconscious() checks for stun, paralysis and resting all in one

**Comment by clusterfack** [comment]
I'm somewhat concerned that you wouldn't be able to leave a sleeper by yourself if you have handcuffs on thanks to the restrained() check

**Comment by Steelcap** [comment]
It doesn't. As per mob_helpers.dm

/mob/proc/isUnconscious() //Returns 1 if unconscious, dead or faking death
if(stat || (status_flags & FAKEDEATH))
return 1

I also had some misgivings about the restrained check, but it seems reasonable.

**Comment by clusterfack** [comment]
If the restrained check remains then there needs to be a way to resist out of sleepers otherwise it becomes another way to permanently put someone inside it

**Pull request merged**